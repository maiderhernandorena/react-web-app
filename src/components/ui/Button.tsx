import React from 'react';
import Text from './Text'

type ButtonProps = {
  onPress: () => void
  label?: string
}

const Button: React.FC<ButtonProps> = ({ label, onPress, children }) => {
  if (!children && !label) {
    return null
  }

  return (
        <button onClick={onPress}>
          {label && <Text message={label} />}
          {children}
        </button>
  );
}

export default Button;
