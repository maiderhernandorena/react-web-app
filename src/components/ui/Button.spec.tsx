import React from 'react';
import Renderer, { act } from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import Button from './Button';

describe('Button Component', () => {

  describe('Snapshots', () => {
    it('Should render correctly with no label', () => {
      const mockFunc = jest.fn()

      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Button onPress={mockFunc} >
          <p>Any children</p>
        </Button>
      );
      expect(wrapper).toMatchSnapshot();
    });
    
    it('Should render correctly with no children', () => {
      const mockFunc = jest.fn()

      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Button onPress={mockFunc} label='Press me'/>
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('Should return null with no label and chidlren', () => {
      const mockFunc = jest.fn()

      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Button onPress={mockFunc} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});