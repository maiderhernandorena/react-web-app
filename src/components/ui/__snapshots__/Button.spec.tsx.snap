// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshots Should render correctly with no children 1`] = `
<button
  onClick={[MockFunction]}
>
  <Text
    message="Press me"
    type="p"
  />
</button>
`;

exports[`Button Component Snapshots Should render correctly with no label 1`] = `
<button
  onClick={[MockFunction]}
>
  <p>
    Any children
  </p>
</button>
`;

exports[`Button Component Snapshots Should return null with no label and chidlren 1`] = `null`;
